tests/test_util.ts

```diff
@@ -1,6 +1,105 @@
-import { Redis, connect, RedisConnectOptions } from "../redis.ts";
-import { assert } from "../vendor/https/deno.land/std/testing/asserts.ts";
+import { connect, Redis, RedisConnectOptions } from "../redis.ts";
 import { delay } from "../vendor/https/deno.land/std/async/mod.ts";
+import { assert } from "../vendor/https/deno.land/std/testing/asserts.ts";
+
+type TestFunc = () => void | Promise<void>;
+type TestServer = {
+  path: string;
+  process: Deno.Process;
+};
+
+export class TestSuite {
+  private encoder = new TextEncoder();
+
+  private tests: { name: string; func: TestFunc }[] = [];
+  private beforeEachs: TestFunc[] = [];
+  private afterAlls: TestFunc[] = [];
+
+  constructor(private prefix: string) {}
+
+  startRedis = async ({
+    port = 6379,
+    clusterEnabled = false,
+  }): Promise<TestServer> => {
+    const path = `testdata/${port}`;
+
+    if (!(await exists(path))) {
+      Deno.mkdirSync(path);
+      Deno.copyFileSync(`testdata/redis.conf`, `${path}/redis.conf`);
+
+      let config = `dir ${path}\nport ${port}\n`;
+      config += clusterEnabled ? "cluster-enabled yes" : "";
+
+      Deno.writeFileSync(`${path}/redis.conf`, this.encoder.encode(config), {
+        append: true,
+      });
+    }
+
+    const process = Deno.run({
+      cmd: ["redis-server", `testdata/${port}/redis.conf`],
+      stdin: "null",
+      stdout: "null",
+    });
+
+    // Ample time for server to finish startup
+    await delay(500);
+    return { path, process };
+  };
+
+  connect(port: number): Promise<Redis> {
+    return connect({ hostname: "127.0.0.1", port });
+  }
+
+  beforeEach(func: TestFunc): void {
+    this.beforeEachs.push(func);
+  }
+
+  afterAll(func: TestFunc): void {
+    this.afterAlls.push(func);
+  }
+
+  test(name: string, func: TestFunc): void {
+    this.tests.push({ name, func });
+  }
+
+  runTests = async (): Promise<void> => {
+    const promises: Promise<void>[] = [];
+
+    this.tests.forEach((test) => {
+      const promise = test.func();
+      promises.push(Promise.resolve(promise));
+
+      Deno.test(`[${this.prefix}] ${test.name}`, () => {
+        this.beforeEachs.forEach(async (f) => await f());
+        return promise;
+      });
+    });
+
+    try {
+      await Promise.allSettled(promises);
+    } finally {
+      this.afterAlls.forEach(async (f) => await f());
+    }
+  };
+
+  teardown(server: TestServer): void {
+    Deno.removeSync(server.path, { recursive: true });
+    server.process.close();
+  }
+}
+
+async function exists(filename: string): Promise<boolean> {
```

Comment:
Is this function adopted from `std/fs/exists.ts`?
If so, please add the copyright header.
https://deno.land/manual/contributing/style_guide

Reply:
uh no I wrote it. But now seeing there is something like this should I use the std version?